src/rand.jl

```diff
@@ -0,0 +1,58 @@
+import Random
+
+function Random.Sampler(
+    RNG::Type{<:Random.AbstractRNG},
+    st::Random.SamplerType{TOrRef},
+    n::Random.Repetition,
+) where {TOrRef<:Union{Arf,ArfRef,Arb,ArbRef,Acb,AcbRef}}
+    T = _nonreftype(TOrRef)
+    # return Random.SamplerSimple(
+    #     Random.SamplerType{T}(),
+    #     Random.SamplerBigFloat(precision(T)),
+    # )
+    res = setprecision(BigFloat, precision(T)) do
+        Random.SamplerSimple(Random.SamplerType{T}(), Random.Sampler(RNG, BigFloat, n))
+    end
+    return res
+end
+
+Random.rand(rng::Random.AbstractRNG, sp::Random.SamplerSimple{A,B,Arf}) where {A,B} =
+    Arf(rand(rng, sp.data), prec = sp.data.prec)
+Random.rand(rng::Random.AbstractRNG, sp::Random.SamplerSimple{A,B,Arb}) where {A,B} =
+    Arb(rand(rng, sp.data), prec = sp.data.prec)
+Random.rand(rng::Random.AbstractRNG, sp::Random.SamplerSimple{A,B,Acb}) where {A,B} =
+    Acb(rand(rng, sp.data), rand(rng, sp.data), prec = sp.data.prec)
+
+function Random.rand(
+    rng::Random.AbstractRNG,
+    st::Random.SamplerTrivial{A,A},
+) where {A<:ArfOrRef}
+    # TODO: Arf(rand(rng, Random.SamplerBigFloat(precision(st[]))))
+    x = setprecision(BigFloat, precision(st[])) do
+        rand(rng, BigFloat)
```

Comment:
I guess you are working around the bug in `Random` which doesn't create a `BigFloat` according to the precision set in the `Sampler`? But I guess you could still store the `SamplerBigFloat` as a `.data` field of a `SamplerSimple`, and do 
```
x = setprecision(BigFloat, sp.data.prec) do
        rand(rng, sp.data)
end
```
(but anyway, the random generation is quite more expensive that the `Sampler` creation`, so the benefit would not be huge).

Reply:
that's what TODO says, but at the moment I don't know how to easily create `SamplerBigFloat` with the given precision ;) any advice on this?

Reply:
Yep:
```
setprecision(BigFloat, prec) do
    Random.Sampler(RNG, BigFloat)
end
```
or 
```
Random.SamplerBigFloat{Random.CloseOpen01{BigFloat}}(prec)
```